### ISSUE TITLE
Show slow tests

### ISSUE THREAD

**Issue by mvorisek** [PHP-CS-Fixer/PHP-CS-Fixer#4728]
This is a feature request to show slow tests like phpunit with ` johnkary/phpunit-speedtrap`.

Usecase: to quickly identify issues like #4656

**Comment by kubawerlos** [comment]
@mvorisek how would you imagine to be done?

**Comment by mvorisek** [comment]
@kubawerlos Very simple, measure ruleset apply time and then report the ruleset name, file name and the elapsed time if the elapsed time is greater than some treshold.

Note, that this feature request is for reporting **AFTER** ruleset is executed. To discover rulesets that never finish or take very long time use verbose log, as there is no way (in single non-async thread) to display some message during the ruleset execution. For verbose logging, see #4372 

What do you think?

**Comment by kubawerlos** [comment]
Rulesets or just rules?

I was thinking once about adding something here: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.1/src/Runner/Runner.php#L150

Maybe some events before starting fixing a file and after finishing.

**Comment by mvorisek** [comment]
> Rulesets or just rules?
You are right, rule not ruleset (I also updated the comment above to prevent any misunderstanding.)

**Comment by mvorisek** [comment]
It seems that verbose level (like `--verbose=3` or `-vvv`) is completely unsupported, even if documented in:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.16/doc/usage.rst

`--verbose=3` ends up with error:
```
In ArgvInput.php line 204:

  [Symfony\Component\Console\Exception\RuntimeException]
  The "--verbose" option does not accept a value.
```

`-vvv` seems to no affect the verbosity at all, `-vvvvvvvvvvv` does not hep too

**Comment by mvorisek** [comment]
closing as performance issues are currently rarely problematic

**Comment by mvorisek** [comment]
I am reopening this issue as when introducing PHP CS Fixer to some larger project it has to be rerun localy many times and if I would be able to list slow rules, it would help - save developer time - to disable them temporary.

**Comment by Wirone** [comment]
I've renamed the issue as it has nothing to do with tests 😉.

**Comment by mvorisek** [comment]
not stale
